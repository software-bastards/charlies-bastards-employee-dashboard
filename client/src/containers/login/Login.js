import React from "react";
import { useForm } from "react-hook-form";
import loginHelper from "../../services/loginHelper";
import { connect, useDispatch } from "react-redux";
import { createSession } from "../../reducers/actions/index";
import { withRouter, useHistory } from "react-router-dom";
function Login() {
  const [message, setMessage] = React.useState("");
  const { register, errors, handleSubmit } = useForm();
  const dispatch = useDispatch();
  const history = useHistory();
  /**
   * @function onSubmit
   * @param {string} data -Values passed in the input
   * @param {*} e - event
   */

  const onSubmit = (value, e) => {
    e.preventDefault();
    loginHelper(value.email, value.password)
      .then((res) =>
        dispatch(
          createSession(
            res.data.message,
            res.data.token,
            res.data.firstname,
            res.data.lastname,
            res.data.id
          )
        )
      )
      .then(history.push("/dashboard"));
    /*       .catch((err) => setMessage(`${err.response.data.message.message}`));
     */
  };

  return (
    <main>
      <form data-testid="form-component" onSubmit={handleSubmit(onSubmit)}>
        <p>{message}</p>
        <label data-testid="test-label" htmlFor="email">
          E-mail
        </label>
        <input
          data-testid="input-form-email"
          type="text"
          name="email"
          ref={register({ required: true })}
        />
        {errors.email && "This field is required"}

        <label data-testid="test-label" htmlFor="password">
          Password
        </label>
        <input
          data-testid="input-form-password"
          type="password"
          name="password"
          ref={register({ required: true })}
        />
        {errors.password && "This field is required"}

        <button data-testid="submit-button" type="submit">
          {" "}
          Login{" "}
        </button>
      </form>
      <button
        onClick={() => (window.location = "http://localhost:5000/auth/google")}
      >
        {" "}
        Google +
      </button>
    </main>
  );
}

export default withRouter(Login);
