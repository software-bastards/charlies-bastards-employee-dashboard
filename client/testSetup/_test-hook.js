import React from "react";
import { shallow, mount } from "enzyme";

export default function testHook(runHook, flushEffects = true) {
  function HookWrapper() {
    const output = runHook();

    return <span output={output} />;
  }

  const wrapperFunc = flushEffects ? mount : shallow;
  const wrapper = wrapperFunc(<HookWrapper />);

  return wrapper.find("span").props().output;
}
