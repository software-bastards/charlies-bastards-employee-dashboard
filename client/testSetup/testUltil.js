import rootReducer from "../src/reducers";
import { createStore } from "redux";

/**
 * @function storeFactory
 * @param {object} initialState  -initial State for the store
 * @returns {Store}- Redux Store
 */

export const storeFactory = (initialState) => {
  return createStore(rootReducer, initialState);
};

/**
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {ShallowWrapper}
 * @param {string} val - Value of data-test atribute for search
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

export const initialState = {
  state: [],
  reducer: {},
};

/**
 * @function expectTruthy - test if any object existes in our component
 * @param {*} receiveObj
 */

export const expectTruthy = (receiveObj) => {
  expect(receiveObj).toBeTruthy() && expect(receiveObj).not.toBeNull();
};
