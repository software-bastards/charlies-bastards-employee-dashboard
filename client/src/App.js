import React from "react";
import Login from "./containers/login/Login";
import Dashboard from "./containers/dashboard/Dashboard";
import { Route, Switch } from "react-router-dom";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import DisplayHours from "./containers/displayhours/DisplayHours";

function App(isAuthenticated) {
  return (
    <div data-test="component-app">
      <Switch>
        <Route exact path="/">
          {!isAuthenticated ? <Redirect to="/dashboard" /> : <Login />}
        </Route>
        <Route path="/dashboard">
          {!isAuthenticated ? <Redirect to="/" /> : <Dashboard />}
        </Route>
        <Route path="/displayhours">
          {!isAuthenticated ? (
            <Redirect to="/displayhours" />
          ) : (
            <DisplayHours />
          )}
        </Route>
      </Switch>
    </div>
  );
}
function mapToProps(state) {
  return { isAuthenticated: state.authorization.token };
}
export default connect(mapToProps)(App);
