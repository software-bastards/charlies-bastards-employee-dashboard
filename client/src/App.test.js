import React from "react";
import App from "./App";
import { Provider } from "react-redux";
import { initialState, expectTruthy } from "../testSetup/testUltil";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import axios from "axios";
import Login from "./containers/login/Login";

test("full app rendering withou error", () => {
  const state = {
    authorization: {
      token: "something",
      id: "someth",
      email: "sth",
      firstname: "sth",
      lastname: "sth",
    },
    data: "2:41:16 PM",
  };
  const historyMock = { push: jest.fn() };
  const mockStore = configureStore();
  const store = mockStore(state);
  const renderComponent = () =>
    render(
      <Router>
        <Provider store={store}>
          <App history={historyMock} />
        </Provider>
      </Router>
    );
  const wrapper = renderComponent();
  expect(wrapper).toMatchSnapshot();
});
