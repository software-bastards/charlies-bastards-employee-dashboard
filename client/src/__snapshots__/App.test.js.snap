// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full app rendering withou error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-app"
      >
        <div
          class="dash-container"
          style="opacity: 0;"
        >
          <div>
            <div
              class="clock"
            >
              11:04:11 AM
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-sm"
              >
                Welcome, 
                sth
                 
                <img
                  alt="dymion"
                  class="avatar"
                  src="https://frontendbastards.nl/images/fe/team/dymion.svg"
                />
              </div>
              <div
                class="col-sm"
              >
                 Worked 
                 hours last month
              </div>
              <div
                class="col-sm"
              >
                Why even 
              </div>
            </div>
          </div>
          <button
            class="btn-dash"
            data-testid="test-display-router"
          >
            Display Your Hours
          </button>
          <button
            class="btn-dash"
            data-testid="test-insert-router"
          >
            Insert New Log
          </button>
          <button
            class="btn-dash"
            data-testid="test-edit-router"
          >
            Edit Your Hours
          </button>
          <button
            class="btn-dash"
            data-testid="test-upload-router"
          >
            Upload Image
          </button>
          <br />
          <button
            class="btn-logout"
            data-testid="test-logout"
          >
             
            Log Out
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-app"
    >
      <div
        class="dash-container"
        style="opacity: 0;"
      >
        <div>
          <div
            class="clock"
          >
            11:04:11 AM
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-sm"
            >
              Welcome, 
              sth
               
              <img
                alt="dymion"
                class="avatar"
                src="https://frontendbastards.nl/images/fe/team/dymion.svg"
              />
            </div>
            <div
              class="col-sm"
            >
               Worked 
               hours last month
            </div>
            <div
              class="col-sm"
            >
              Why even 
            </div>
          </div>
        </div>
        <button
          class="btn-dash"
          data-testid="test-display-router"
        >
          Display Your Hours
        </button>
        <button
          class="btn-dash"
          data-testid="test-insert-router"
        >
          Insert New Log
        </button>
        <button
          class="btn-dash"
          data-testid="test-edit-router"
        >
          Edit Your Hours
        </button>
        <button
          class="btn-dash"
          data-testid="test-upload-router"
        >
          Upload Image
        </button>
        <br />
        <button
          class="btn-logout"
          data-testid="test-logout"
        >
           
          Log Out
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
